src/main/java/com/metamx/http/client/pool/ResourcePool.java

```diff
@@ -166,9 +172,18 @@ V get()
         if (closed) {
           log.info(String.format("get() called even though I'm closed. key[%s]", key));
           return null;
-        } else if (!objectList.isEmpty()) {
-          poolVal = objectList.removeFirst();
-        } else if (deficit > 0) {
+        } else if (!objectMap.isEmpty())
+        {
+          Map.Entry<Long,V> firstEntry = objectMap.pollFirstEntry();
+          if(System.nanoTime() - firstEntry.getKey() > maxTimeInNanosToHold)
```

Comment:
formatting

Reply:
It doesn't work the way that @himanshug explained here: https://github.com/druid-io/druid/issues/3393#issuecomment-345005807 it removes an object upon request, not by timeout. Is that intended?